### PR TITLE
Hrcpp 475/get cache names

### PR DESCRIPTION
**Pull request by rigazilla**
RemoteCacheManager.getCacheNames() implementation

https://issues.jboss.org/browse/HRCPP-475

### CODE REVIEW

**Comment by alanfx** [comment]
This one needs a rebase now, @rigazilla

**Comment by rigazilla** [comment]
rebased, @alanfx

File: jni/src/test/java/org/infinispan/client/jni/hotrod/JniTest.java [hunk 1]
```diff
@@ -98,8 +98,7 @@ public static void main(String[] args) {
             "GetAllLocalTest.testBulkGetAfterLifespanExpire",
             "GetAllReplTest.testBulkGetAfterLifespanExpire",
             "RemoteCacheAdminTest.cacheReindexTest",                             // See HRCPP-469
-            "RemoteCacheAdminTest.cacheCreateWithEmbeddedConfigurationTest",     // conf to XML doesn't work here
-            "RemoteCacheAdminTest.testGetCacheNames"                             // HRCPP-475
```
Comment (alanfx) [inline jni/src/test/java/org/infinispan/client/jni/hotrod/JniTest.java]: This test can't be used? Are there any other tests for this?

Comment (rigazilla) [reply jni/src/test/java/org/infinispan/client/jni/hotrod/JniTest.java]: This test needs the configuration to XML feature which is not implemented in C++ or C#

**Comment by alanfx** [comment]
@rigazilla Looks good. I got confused because you removed `RemoteCacheAdminTest.testGetCacheNames` from the failing tests! Merging

**Pull request merged**